File: 1768_merge_strings_alternately/main.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        word = ''
        while word1 or word2:
            if word1:
                word += word1[:1]
                word1 = word1[1:]
            if word2:
                word += word2[:1]
                word2 = word2[1:]
        return word

solution = Solution()
print(solution.mergeAlternately(word1="ab", word2="pqrs"))

File: 1768_merge_strings_alternately/main.go
package main

import "fmt"

func mergeAlternately(word1 string, word2 string) string {
	word := ""
	for i := 0; (i < len(word1)) || (i < len(word2)); i++ {
		if word1 != "" {
			word1 = word1[0:]
			word += word1[:1]
		}
		if word2 != "" {
			word2 = word2[0:]
			word += word2[:1]
		}
	}
    return word
}

func main() {
	word1 := "abc"
	word2 := "pqr"
	word := mergeAlternately(word1, word2)
	fmt.Println(word)
}